ui/framework/src/appui-react/UiFramework.ts

```diff
@@ -185,7 +185,7 @@ export class UiFramework {
 
     const oidcClient = IModelApp.authorizationClient;
     // istanbul ignore next
-    if (isFrontendAuthorizationClient(oidcClient)) {
+    if (isBrowserAuthorizationClient(oidcClient)) {
```

Comment:
We won't be able to rely on this for long either. We're going to need some other way of handling this event or just removing it altogether. Can you investigate what that would look like?

Reply:
I've been looking for a while now and I'm not aware of a good solution yet. The listener being added is for when the token expires or the AuthClient is disposed of, and I don't know of another way of knowing when this happens with the current AuthorizationClient interface. There are of course ways of keeping the same workflow and avoid checking for the particular type of class, but from my point of view it really needs that even handler.

I'll keep digging and see if there's any alternative I can find...

Reply:
@bsteinbk the `onUserStateChanged` event is being used to clear all of the UI component. Is that due to the desire to return the app back to the initial state whenever someone signs out or gets kicked out?

Reply:
We still need to figure out what to do here about this event. We don't want to ui-framework taking on this dependency so we'll need to find a way around it.

Reply:
What if we just checked if the `oidcClient` has the property of `onUserStateChanged`? That would mean any implementation of AuthorizationClient that doesn't implement it would not perform the action.

Reply:
@calebmshafer - @NancyMcCallB and I discussed this and we don't exactly know the history of why Dan added it, but we see that it really just closes all popups windows, dialogs and messages it doesn't do much else.  So there is no reason that an app could not monitor the user change themselves and call ConfigurableUiManager.closeUi() themselves.  We would just need to make that method public and document it in NextVersion.md just in case someone expects it to close automatically.